unit-tests/src/test/java/com/novoda/merlin/MerlinsBeardShould.java

```diff
@@ -53,4 +54,27 @@ public void returnTrueForIsConnectedWhenNetworkConnectionIsAvailable() {
         assertThat(merlinsBeard.isConnected()).isTrue();
     }
 
-}
\ No newline at end of file
+    @Test
+    public void returnTrueForIsConnectedToWifiWhenNetworkConnectedToWifiIsConnected() {
+        when(mockConnectivityManager.getNetworkInfo(WIFI.getValue())).thenReturn(mockNetworkInfo);
+        when(mockNetworkInfo.isConnected()).thenReturn(true);
+
+        assertThat(merlinsBeard.isConnectedTo(WIFI)).isTrue();
+    }
+
+    @Test
+    public void returnFalseForIsConnectedToWifiWhenNetworkConnectedToWifiIsNotConnected() {
```

Comment:
It would be good to have a test with getActiveNetworkInfo() that returns true and for wifi false, to show that the activeNetworkInfo is not used.